### PR TITLE
Module based docgen (including troy modules)

### PR DESCRIPTION
**Pull request by mfelsche**
# Pull request

## Description

Generate stdlib docs from modules, also include troy modules now.

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

### CODE REVIEW

File: tests/flow_error.rs [hunk 1]
```diff
@@ -53,7 +53,9 @@ macro_rules! test_cases {
 
                 match parse(&contents) {
                     Err(e) => {
-                        let got = Dumb::error_to_string(&e)?;
```
Comment (Licenser) [inline tests/flow_error.rs:56]: Why do we spell out the highlighter generation?

File: tremor-cli/src/doc.rs [hunk 2]
```diff
@@ -12,65 +12,167 @@
...
+use tremor_script::arena::Arena;
+use tremor_script::module::{Id, Module};
 
+fn push_line(line: &str, buf: &mut String) {
+    buf.push_str(line);
+    if !line.ends_with('\n') {
+        buf.push('\n');
+    }
+}
+#[allow(clippy::too_many_lines)]
```
Comment (Licenser) [inline tremor-cli/src/doc.rs:30]: Could we refactor this to not require an allow?

File: tests/flow_error.rs (hunk 1 above)
Comment (mfelsche) [reply tests/flow_error.rs:56]: What exactly do you mean?

Comment (Licenser) [reply tests/flow_error.rs:56]: this seems to unroll `error_to_string` and I wondered why

Comment (mfelsche) [reply tests/flow_error.rs:56]: It covers the uncovered function `Deploy::format_error_with`. Sneaky, i know!

Comment (Licenser) [reply tests/flow_error.rs:56]: alternatively, we could just `format_error_with`? it seems to be a copy of `format_error`. This feels like a harmful example of code for coverage's sake :(

Comment (Licenser) [reply tests/flow_error.rs:56]: `format_error_with`, as far as I can tell, is never called anywhere so removing it seems the right thing to do.

File: src/connectors/impls/stdio.rs [hunk 3]
```diff
@@ -34,6 +34,7 @@ lazy_static! {
             let mut stream = stdin();
             let mut buffer = [0_u8; INPUT_SIZE_BYTES];
             while let Ok(len) = stream.read(&mut buffer).await {
+                dbg!(len);
```
Comment (Licenser) [inline src/connectors/impls/stdio.rs:37]: weeeh I found a dbg :)
```suggestion
```

**Review by darach — APPROVED** [review]
LGTM 👍

**Review by Licenser — APPROVED** [review]
🚀

**Pull request merged**